packages/ripple-keypairs/src/index.ts

```diff
@@ -1,33 +1,41 @@
-import brorand = require('brorand')
-import * as hashjs from 'hash.js'
-import * as elliptic from 'elliptic'
-
 import * as addressCodec from 'ripple-address-codec'
-import { derivePrivateKey, accountPublicFromPublicGenerator } from './secp256k1'
-import * as utils from './utils'
 
-const Ed25519 = elliptic.eddsa('ed25519')
-const Secp256k1 = elliptic.ec('secp256k1')
+import { secp256k1 as nobleSecp256k1 } from '@noble/curves/secp256k1'
+import { ed25519 as nobleEd25519 } from '@noble/curves/ed25519'
+import { numberToBytesBE } from '@noble/curves/abstract/utils'
+import { ripemd160 } from '@xrplf/isomorphic/ripemd160'
+import { sha256 } from '@xrplf/isomorphic/sha256'
+import { bytesToHex, hexToBytes, randomBytes } from '@xrplf/isomorphic/utils'
+
+import { accountPublicFromPublicGenerator, derivePrivateKey } from './secp256k1'
+import Sha512 from './Sha512'
+import assert from './assert'
+
+export type ByteArray = number[] | Uint8Array
+export type HexString = string
+type Input = string | Uint8Array
 
-const { hexToBytes } = utils
-const { bytesToHex } = utils
+const SECP256K1_PREFIX = '00'
+
+function hash(message: Input | number[]): Uint8Array {
+  return new Sha512().add(message).first256()
+}
 
 function generateSeed(
   options: {
     entropy?: Uint8Array
     algorithm?: 'ed25519' | 'ecdsa-secp256k1'
   } = {},
 ): string {
-  if (!(!options.entropy || options.entropy.length >= 16)) {
-    throw new Error('entropy too short')
-  }
-  const entropy = options.entropy ? options.entropy.slice(0, 16) : brorand(16)
+  assert.ok(
```

Comment:
Weren't we trying to avoid using assert in favor of throwing errors directly?

Reply:
We are.  In this case sublimator created a utility to make the code look cleaner.